### PR TITLE
Feature/messages router

### LINKED ISSUE #107: Create Messages Router
**__User Story Description__**
As a [role]
I want to [action to be performed]
So I can [result to be achieved]

**__Steps to Follow (optional)__**
- [ ] GET route - get messages
- [ ] POST route - post messages

**__Additional Considerations__**
Screen Shot 2022-04-27 at 23 03 37

### PR DESCRIPTION
**Pull request by moody2times**
# Message Router

I have created the message router functionality. However, I do not understand the details of the implementation you have in mind. Please, feel free to add code sample to your review.

closes #107

### CODE REVIEW

File: server/Controller/messageController.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+      message: 'User_id field is required!',
+      statusCode: 400,
+    })
+    next(error)
+  }
+
+  // if there is an id, then
+  try {
+    // query the database for a match
+    const result = await Message.find({ sender: id, reciever: id })
```
Comment (KhadidjaArezki) [inline server/Controller/messageController.js]: The find method here will look for a document where the sender field is equal to id AND the receiver is equal to id. As a massage cannot be sent and received by the same user, this query should always return nothing.
This is what I had in mind:
`const result = await Message.find().or([{ sender: id }, { receiver: id }])`
Where the collection is queried for messages where the sender id is equal to id OR the receiver id is equal to id

File: server/Controller/messageController.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    // if there is a match, send 200 status with result
```
Comment (KhadidjaArezki) [inline server/Controller/messageController.js:40]: Before returning the results, here is what I was suggesting we do:
1. Get the current user's list of contacts:
`const friends = await User.findById(id).contacts`
2. Create a chats array:
`chats= []`
3. Loop over each friend_id in the friends array and look for messages where sender === friend_id OR receiver === friend_id:
`for (friend_id in friends) { 
     const friendMessages = result.filter(message=> 
        message.sender === friend_id || message.receiver === friend_id 
 ) 
 chats.concat({ friend_id, messages: friendMessages })`
 4. return chats

**Review by KhadidjaArezki — CHANGES_REQUESTED** [review]

File: server/Controller/messageController.js (hunk 1 above)
Comment (moody2times) [reply server/Controller/messageController.js]: > The find method here will look for a document where the sender field is equal to id AND the receiver is equal to id. As a massage cannot be sent and received by the same user, this query should always return nothing.

Yeah, it was just a placeholder until I understand what you want me to implement.

File: server/Controller/messageController.js (hunk 2 above)
Comment (moody2times) [reply server/Controller/messageController.js:40]: >    `const friends = await User.findById(id).contacts`

Is it not an array of contacts that is stored in `friends`? If it is an array, the `for-in` loop should not work.

Comment (KhadidjaArezki) [reply server/Controller/messageController.js:40]: Yeah, it was just pseudo-code, I meant loop through the array

**Review by KhadidjaArezki — APPROVED** [review]
Great work as always.

**Pull request merged**